src/stacked_borrows/mod.rs

```diff
@@ -973,6 +973,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
 
             fn visit_value(&mut self, place: &PlaceTy<'tcx, Provenance>) -> InterpResult<'tcx> {
+                // We have nothing to do for ZSTs. They contains no bytes so they cannot store
+                // pointers, but they can contain way more fields than we can ever visit.
+                if !place.layout.is_unsized()
+                    && place.layout.size < Primitive::Pointer.size(self.ecx)
```

Comment:
`self.ecx.pointer_size()` should work, possibly after importing [this trait](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_const_eval/interpret/trait.PointerArithmetic.html).